drizzlepac/hlautils/se_source_generation.py

```diff
@@ -387,46 +412,119 @@ def write_catalog(seg_cat, catalog_filename, product="tdp"):
-        seg_table.rename_column("source_sum", "flux")
-        seg_table.rename_column("source_sum_err", "flux_err")
-        seg_table.rename_column("cxx", "cxx_image")
-        seg_table.rename_column("cxy", "cxy_image")
-        seg_table.rename_column("cyy", "cyy_image")
-        seg_table.rename_column("covar_sigx2", "x2_image")
-        seg_table.rename_column("covar_sigy2", "y2_image")
-        seg_table.rename_column("covar_sigxy", "xy_image")
-
-        # Write out the official total detection product source catalog
-        seg_table["x_image"].info.format = ".10f"
-        seg_table["y_image"].info.format = ".10f"
+        seg_table["RA_icrs"].description = "SExtractor Column RA"
+        seg_table["Dec_icrs"].description = "SExtractor Column Dec"
+        seg_table["RA_icrs"].unit = u.deg
+        seg_table["Dec_icrs"].unit = u.deg
+
+        # Add a description for columns which map to SExtractor catalog columns
+        seg_table["xcentroid"].description = "SExtractor Column x_image "
+        seg_table["ycentroid"].description = "SExtractor Column y_image "
+        seg_table["background_at_centroid"].description = "SExtractor Column background"
+        seg_table["source_sum"].description = "SExtractor Column flux_iso"
+        seg_table["source_sum_err"].description = "SExtractor Column fluxerr_iso"
+        # FIX: is mapping to _image or _world?  _image
+        seg_table["cxx"].description = "SExtractor Column cxx_image, ellipse parameter"
+        seg_table["cyy"].description = "SExtractor Column cyy_image, ellipse parameter"
+        seg_table["cxy"].description = "SExtractor Column cxy_image, ellipse parameter"
+        # FIX: is the mapping to _image or _world?
+        seg_table["covar_sigx2"].description = "SExtractor Column x2_image, (0,0) element of covariance matrix"
+        seg_table["covar_sigy2"].description = "SExtractor Column y2_image, (1,1) element of covariance matrix"
+        seg_table["covar_sigxy"].description = "SExtractor Column xy_image, (0,1) and (1,0) elements of covariance matrix"
+
+        seg_table["xmin"].description = "SExtractor Column xmin_image"
+        seg_table["xmax"].description = "SExtractor Column xmax_image"
+        seg_table["ymin"].description = "SExtractor Column ymin_image"
+        seg_table["ymin"].description = "SExtractor Column ymax_image"
+
+        # Write out the official filter detection product source catalog
+        seg_table["xcentroid"].info.format = ".10f"
+        seg_table["ycentroid"].info.format = ".10f"
         seg_table["RA_icrs"].info.format = ".10f"
         seg_table["Dec_icrs"].info.format = ".10f"
-        seg_table["RA_icrs"].unit = u.deg
-        seg_table["Dec_icrs"].unit =  u.deg
         print("seg_table (filter): {}".format(seg_table))
 
         seg_table.write(catalog_filename, format="ascii.ecsv")
         log.info("Wrote filter source catalog: {}".format(catalog_filename))
 
+def get_header_data(imghdu):
+
+    keyword_dict = {}
+
```

Comment:
General comment for this entire function: using the exact metadata terms for the dictionary keys would make applying this dict to the metadata almost trivial for those keywords that have a 1-to-1 relation with the FITS keywords.  For example, 'Proposal ID' instead of 'proposal_id'.